scarb/src/core/package/name.rs

```diff
@@ -88,9 +88,89 @@ impl PackageName {
             }
         }
 
+        if PackageName::is_keyword(&name) {
+            bail!(
+                "the name `{name}` cannot be used as a package name, \
+                names can't use cairo keywords see the full list at \
+                https://docs.cairo-lang.org/language_constructs/keywords.html and \
+                https://github.com/starkware-libs/cairo/blob/dc7f0d5ca76b7409816d99772033abf58975a872/crates/cairo-lang-parser/src/lexer.rs#L184-L209"
+            )
+        }
+
+        if ["test", "scarb"].contains(&name.as_ref()) {
+            bail!("the name `{name}` cannot be used as a package name")
+        }
```

Comment:
Why? It seems to be totally fine to be able to make such packages. I can even imagine us defining such ones in the future.

Reply:
I wanted to stop users from creating such packages, especially considering the introduction of package registry. It might become confusing and problematic, but probably there is a better solution to this problem.

Reply:
I don't think we should stop people from doing such stuff, but we can allow ourselves to stop them from **publishing** it. So that's registry problem to consider